src/solver/lu_decomposition.cu

```diff
@@ -85,6 +87,11 @@ namespace micm
               size_t A_idx = d_aki[aki_offset++] + tid;
               d_L[L_idx] = d_A[A_idx];
             }
+            else {
+              size_t L_idx = d_lki_nkj[lki_nkj_offset].first + tid;
+              d_L[L_idx] = 0;
+            }
+
             for (size_t ikj = 0; ikj < d_lki_nkj[lki_nkj_offset].second; ++ikj)
             {
               size_t L_idx_1 = d_lki_nkj[lki_nkj_offset].first + tid;
```

Comment:
I think this line can be deleted as well

Reply:
fixed

Reply:
oh I was suggesting for line `94-101`
```C++
  for (size_t ikj = 0; ikj < d_lki_nkj[lki_nkj_offset].second; ++ikj)
  {
    size_t L_idx_1 = d_lki_nkj[lki_nkj_offset].first + tid;   // Delete this 
    ...
    d_L[L_idx] -= d_L[L_idx_2] * d_U[U_idx];    // Update d_L[L_idx_1] to d_L[L_idx] 
    ++lkj_uji_offset;
  }
```